src/common/conditional_compilation/docs/develop_cc_for_new_component.md

```diff
@@ -0,0 +1,238 @@
+
+https://github.com/openvinotoolkit/openvino/blob/34b76584f724ad07ed27c8ea27777ac46df92c23/src/common/conditional_compilation/include/openvino/cc/selective_build.h#L181-L183
+
+There is an example of `conditional_compilation_gen.h`:
+```
+#define ov_pass_FixRtInfo_run_on_function 1
+#define ov_pass_GraphRewrite_run_on_model 1
+#define ov_pass_InitNodeInfo_run_on_function 1
+#define ov_pass_ConstantFolding_run_on_model 1
+#define ov_pass_SubtractFusion 1
+#define ov_pass_SharedShapeOf_run_on_function 1
+#define ov_pass_SimplifyShapeOfSubGraph_run_on_function 1
+#define ov_pass_PropagateNMSPath 1
+#define ov_pass_UselessStridedSliceEraser_run_on_function 1
+#define ov_pass_SharedStridedSliceEraser_run_on_function 1
+#define ov_pass_GroupedStridedSliceOptimizer_run_on_function 1
+#define ov_pass_StridedSliceOptimization_run_on_function 1
+#define ov_pass_EliminateConvert 1
+#define ov_pass_EliminateEltwise 1
+#define ov_pass_Unsupported 1
+#define ov_pass_PassThrough 1
+#define ov_pass_Binary 1
+#define ov_pass_ConvertPassThrough 1
+#define ov_pass_BackwardGraphRewrite_run_on_model 1
+#define ov_pass_Constant_run_on_function 1
+```
+
+#### OV_SWITCH
+It will decide whether the template class parameter match operation need be excluded from final binaries. If the match operation is false in analyzed mode, the code region will be excluded in SELECTIVE_BUILD mode.
+
+#### OV_CASE and OV_CASE2
+They are used with OV_SWITCH to provide match cases.
+
+#### MATCHER_SCOPE
+It defines a string to represent the matcher name in this region, and then decide whether the code region is inactivated and excluded from final binaries, according to the OV_SCOPE result in analyzed mode.
+
+https://github.com/openvinotoolkit/openvino/blob/34b76584f724ad07ed27c8ea27777ac46df92c23/src/common/conditional_compilation/include/openvino/cc/pass/itt.hpp#L30-L33
+
+#### RUN_ON_FUNCTION_SCOPE and RUN_ON_MODEL_SCOPE
+They are very similar, and functionality is same with MATCHER_SCOPE plus throw error.
+
+### 3. Non Conditional Compilation
+It means that conditional compilation is disabled.
+
+#### OV_CC_DOMAINS
+It is empty in this mode
+
+#### OV_SCOPE
+It is empty in this mode
+
+#### OV_SCOPE and OV_CASE
+They will do match operation without itt scope task.
+
+#### MATCHER_SCOPE
+It is empty.
+
+#### RUN_ON_FUNCTION_SCOPE and RUN_ON_MODEL_SCOPE
+They are empty.
+
+
+## Enable conditional complication for a new component or domain
+### step 1: requirement analyze
+Analyze which code region need conditional complication and figure out requirement for conditional complication.
+
+### step 2: create itt.hpp if possible
+Create itt.hpp file in the target component directory, in which you need define some particular macro needed by conditional compilation if existed macro cannot meet requirement. An example [itt.hpp](https://github.com/openvinotoolkit/openvino/blob/master/src/core/src/itt.hpp).
+
+- Add macro into itt.hpp to define new CC domain:
+```
+#pragma once
```

Comment:
Can we use the same approach:
https://github.com/openvinotoolkit/openvino/pull/13852/files#diff-9ff0c0918d285b4c5886415e1b1db4b1d17de017224be73646985befb5ae5073R103

Reply:
done